ql/src/test/results/clientpositive/llap/ctlt_translate_external.q.out

```diff
@@ -0,0 +1,108 @@
+PREHOOK: query: create table test_mm(empno int, name string) partitioned by(dept string) stored as orc tblproperties('transactional'='true', 'transactional_properties'='default')
+PREHOOK: type: CREATETABLE
+PREHOOK: Output: database:default
+PREHOOK: Output: default@test_mm
+POSTHOOK: query: create table test_mm(empno int, name string) partitioned by(dept string) stored as orc tblproperties('transactional'='true', 'transactional_properties'='default')
+POSTHOOK: type: CREATETABLE
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@test_mm
+PREHOOK: query: create table test_external like test_mm
+PREHOOK: type: CREATETABLE
+PREHOOK: Output: database:default
+PREHOOK: Output: default@test_external
+POSTHOOK: query: create table test_external like test_mm
+POSTHOOK: type: CREATETABLE
+POSTHOOK: Output: database:default
+POSTHOOK: Output: default@test_external
+PREHOOK: query: desc formatted test_external
+PREHOOK: type: DESCTABLE
+PREHOOK: Input: default@test_external
+POSTHOOK: query: desc formatted test_external
+POSTHOOK: type: DESCTABLE
+POSTHOOK: Input: default@test_external
+# col_name            	data_type           	comment             
+empno               	int                 	                    
+name                	string              	                    
+	 	 
+# Partition Information	 	 
+# col_name            	data_type           	comment             
+dept                	string              	                    
+	 	 
+# Detailed Table Information	 	 
+Database:           	default             	 
+#### A masked pattern was here ####
+Retention:          	0                   	 
+#### A masked pattern was here ####
+Table Type:         	MANAGED_TABLE       	 
+Table Parameters:	 	 
+	COLUMN_STATS_ACCURATE	{\"BASIC_STATS\":\"true\"}
+	bucketing_version   	2                   
+	numFiles            	0                   
+	numPartitions       	0                   
+	numRows             	0                   
+	rawDataSize         	0                   
+	totalSize           	0                   
+	transactional       	true                
+	transactional_properties	default             
+#### A masked pattern was here ####
```

Comment:
Since the output location is masked out here, we really don't know if the table is pointing to managed or external location. I think it would be nice to have Junit test and then assert on table location. Then we can make sure no regressions happen in the future.

Reply:
Looked at unit tests at TestDatabaseTableDefault that identify the table location, but in managed and external tables both seem to have the same path. Do we have a way to identify or differentiate the paths?

Reply:
Yeah looks like in the TestDatabaseTableDefault class, the metastore.metadata.transformer config is coming as empty and as a result, the transformer is not coming into effect which leaves managed and external paths as same.
Can you use this class https://github.com/apache/hive/blob/master/itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/TestWarehouseExternalDir.java#L175 to test your patch? This class was able to distinguish between managed and external directories.

Reply:
@saihemanth-cloudera Historically this test has been flaky for multiple times and been ignored sometime before. I am reluctant to bring this back. Do you have a different class/test that you used for previous work in this area?

Reply:
@ramesh0201 - There are a couple of other classes that distinguish managed and external locations but those are irrelevant to the current functionality. So I would suggest adding a new method to this class. This new method would use an HMS client that is built based on configs as you mentioned in the above .q file, and use this client to create a table so that the created table is translated to an external table based on the configs, then we can do an assert on the table locations.

Reply:
@saihemanth-cloudera Can you review the latest?